app/src/components/games/reefscape/ReefscapeAutoModal.tsx

```diff
@@ -0,0 +1,438 @@
+        <Text
+          style={{
+            color: colors.text,
+            fontSize: 30,
+            fontWeight: 'bold',
+          }}>
+          AUTO
+        </Text>
+        <View
+          style={{
+            height: '100%',
+            width: '100%',
+            alignItems: 'center',
+            paddingHorizontal: '5%',
+            flex: 1,
+            marginBottom: '5%',
+            display: 'flex',
+            flexDirection: 'column',
+          }}>
+          <Pressable
+            style={{
+              flexDirection: 'row',
+              width: '40%',
+              backgroundColor: colors.border,
+              justifyContent: 'space-evenly',
+              alignItems: 'center',
+              paddingHorizontal: '2%',
+              paddingVertical: '5%',
+              marginBottom: '5%',
+              borderRadius: 10,
+            }}
+            onPress={() => {
+              const lastAction = history.pop();
+              ReactNativeHapticFeedback.trigger('impactLight');
+              if (lastAction) {
+                switch (lastAction.action) {
+                  case 'score':
+                  case 'intake':
+                    setAutoPath(paths =>
+                      paths.filter(path => path.nodeId !== lastAction.pieceID),
+                    );
+                    break;
+                  case 'miss':
+                    setAutoPath(paths =>
+                      paths.map(path =>
+                        path.nodeId === lastAction.pieceID
+                          ? {...path, state: 'success'}
+                          : path,
+                      ),
+                    );
+                    break;
+                  case 'reset':
+                    setAutoPath(paths => [
+                      ...paths,
+                      {
+                        type: ReefscapeActionType.PickupGround,
+                        nodeId: lastAction.pieceID,
+                        order: paths.length,
+                        state: 'success',
+                      },
+                    ]);
+                    break;
+                  case ReefscapeActions[ReefscapeActionType.ScoreProcessor]
+                    .link_name:
+                  case ReefscapeActions[ReefscapeActionType.MissProcessor]
+                    .link_name:
+                  case ReefscapeActions[ReefscapeActionType.MissCoral]
+                    .link_name:
+                  case ReefscapeActions[ReefscapeActionType.ScoreCoralL1]
+                    .link_name:
+                  case ReefscapeActions[ReefscapeActionType.ScoreCoralL2]
+                    .link_name:
+                  case ReefscapeActions[ReefscapeActionType.ScoreCoralL3]
+                    .link_name:
+                  case ReefscapeActions[ReefscapeActionType.ScoreCoralL4]
+                    .link_name:
+                    setArrayData(prevArrayData => {
+                      const {index} = linkItemMap[lastAction.action];
```

Comment:
There is a bug where it seems this returns undefined.
<img width="390" alt="image" src="https://github.com/user-attachments/assets/e009f407-a7b3-4888-9249-38c9a214184d" />